raiden/blockchain/abi.py

```diff
@@ -227,6 +248,20 @@ def instantiate(self):
             )
 
             self.is_instantiated = True
+            self.init_contract_versions()
+
+    def init_contract_versions(self):
+        contracts = [('HumanStandardToken.sol', CONTRACT_HUMAN_STANDARD_TOKEN),
+                     ('ChannelManagerContract.sol', CONTRACT_CHANNEL_MANAGER),
+                     ('EndpointRegistry.sol', CONTRACT_ENDPOINT_REGISTRY),
+                     ('NettingChannelContract.sol', CONTRACT_NETTING_CHANNEL),
+                     ('Registry.sol', CONTRACT_REGISTRY)]
+        for contract_file, contract_name in contracts:
+            self.contract_to_version[contract_name] = parse_contract_version(contract_file)
```

Comment:
Hm, then it'll be required to store the contract version somewhere when creating the release package.

Reply:
Yes, currently this is done using the precompile json file:

https://github.com/raiden-network/raiden/pull/1099/files#diff-303b99a788c9b418295ac84b84782c68R139

Adding the version to the file name or to the serialized json would do.

Reply:
> I believe the smart contract source code is not distributed with the python package. If that is the case, this would fail to execute with an pip install and with the AppImage.

solidity sources are distributed (see `MANIFEST.in`). Until we eventually switch over to using `raiden-contracts` library I'd keep this.